doc/sphinxext/gh_substitutions.py

```diff
@@ -6,6 +6,10 @@
 https://doughellmann.com/blog/2010/05/09/defining-custom-roles-in-sphinx/
 
 """
+
```

Comment:
shouldn't this go to the very very top of every file?

Reply:
I think it should go right below the module docstrings 🤔 that's how it's in sklearn, too, see: https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/decomposition/_base.py

Reply:
ok then!

Reply:
The guideline says "near the beginning" (https://www.bestpractices.dev/en/projects/7783?criteria_level=2)

> The project MUST include a license statement in each source file. This MAY be done by including the following inside a comment near the beginning of each file:

:-) 

I am going ahead and will merge this now. We can always change it later. Thanks!

Reply:
yes all good! thanks for the work!!!